Jint/Native/Array/ArrayInstance.cs

```diff
@@ -8,22 +8,19 @@ namespace Jint.Native.Array
 {
     public class ArrayInstance : ObjectInstance
     {
+        // cache key container for array iteration for less allocations
+        private static readonly ThreadLocal<List<uint>> keyCache = new ThreadLocal<List<uint>>(() => new List<uint>());
+
         private readonly Engine _engine;
-        private IDictionary<uint, PropertyDescriptor> _array = new MruPropertyCache2<uint, PropertyDescriptor>();
+        private readonly Dictionary<uint, PropertyDescriptor> _array = new Dictionary<uint, PropertyDescriptor>();
```

Comment:
MruPropertyCache2 does key.Equals(newKey) checks that will cause boxing with uints, for this use case normal Dictionary<K,V> should be fine, not accessing via interface removes virtual dispatch